seqr/views/utils/json_to_orm_utils.py

```diff
@@ -34,6 +42,16 @@ def _update_model_from_json(model_obj, json, user=None, verbose=False, allow_unk
                 model_obj_name = getattr(model_obj, 'guid', model_obj.__name__)
                 logger.info("Setting {0}.{1} to {2}".format(model_obj_name, orm_key, value))
             setattr(model_obj, orm_key, value)
+            try:
+                if xbrowse_model is not None:
+                    xbrowse_kwargs = convert_seqr_kwargs_to_xbrowse_kwargs(xbrowse_model, **{orm_key: value})
+                    for key, value in xbrowse_kwargs.items():
+                        setattr(model_obj, key, value)
+            except Exception as e:
+                logger.error("Unable to update xbrowse model {0}: {1}".format(xbrowse_model, e))
 
     if modified:
         model_obj.save()
+
+        if xbrowse_model is not None:
+            xbrowse_model.save()
```

Comment:
so this all works, but I have an idea that I think is more readable where instead of setting the fields as we go we keep track of all the fields that are actually getting updated and then just call `update_seqr_model` with those:

```
seqr_update_fields = {}

for json_key, value in json.items():
        orm_key = _to_snake_case(json_key)
        if allow_unknown_keys and not hasattr(model_obj, orm_key):
            continue
        if getattr(model_obj, orm_key) != value:
            if orm_key in model_obj._meta.internal_json_fields and not (user and user.is_staff):
                raise ValueError('User {0} is not authorized to edit the internal field {1}'.format(user, orm_key))
            if verbose:
                model_obj_name = getattr(model_obj, 'guid', model_obj.__name__)
                logger.info("Setting {0}.{1} to {2}".format(model_obj_name, orm_key, value))
            seqr_update_fields[orm_key] = value

    if seqr_update_fields:
        update_seqr_model(model_obj, **seqr_update_fields)
```

Reply:
oh yeah, that's much nicer  👍